core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -718,6 +725,33 @@ private void maybeRefreshEngine() {
         }
     }
 
+    private void maybeUpdateGlobalCheckpoints() {
+        if (indexSettings.getGlobalCheckpointInterval().millis() > 0) {
```

Comment:
Is this guard really needed? The task only gets scheduled if the interval is positive. The only situation this guards against is the setting being dynamically changed to zero or -1from a positive value and a task was already scheduled?